Java/core/src/main/java/com/amazon/randomcutforest/sampler/IStreamSampler.java

```diff
@@ -135,4 +135,16 @@ default boolean isFull() {
      * @return the number of points currently contained by the sampler.
      */
     int size();
+
+    /**
+     * changes the time dependent sampling on the fly
+     * 
+     * @param lambda the rate of sampling
+     */
+    void setLambda(double lambda);
+
+    void setMaxSequenceIndex(long maxSequenceIndex);
+
+    void setSequenceIndexOfMostRecentLambdaUpdate(long index);
```

Comment:
I don't think `setLambda` and `setSequenceIndexOfMostRecentLambdaUpdate` should be in the interface, because lambda is a property of a particular sampling method.

Reply:
Ok. That makes sense. We may have to force object conversion in mappers - but let me try

Reply:
The forest needs access to setLambda. Made it a default function to be overridden.